flow/cmd/handler.go

```diff
@@ -396,15 +393,13 @@ func (h *FlowRequestHandler) FlowStateChange(
 				}
 				changeErr = model.FlowSignalStateChange.SignalClientWorkflow(ctx, h.temporalClient, workflowID, "", req)
 			}
-		case protos.FlowStatus_STATUS_TERMINATING:
-			if currState == protos.FlowStatus_STATUS_COMPLETED {
-				changeErr = h.shutdownFlow(ctx, req.FlowJobName, req.DropMirrorStats, req.SkipDestinationDrop)
-			} else {
-				changeErr = model.FlowSignalStateChange.SignalClientWorkflow(ctx, h.temporalClient, workflowID, "", req)
-			}
-		case protos.FlowStatus_STATUS_TERMINATED: // backwards compat, causes grpc timeouts
-			if currState != protos.FlowStatus_STATUS_TERMINATED {
-				changeErr = h.shutdownFlow(ctx, req.FlowJobName, req.DropMirrorStats, req.SkipDestinationDrop)
+		case protos.FlowStatus_STATUS_TERMINATING, protos.FlowStatus_STATUS_TERMINATED:
+			if currState != protos.FlowStatus_STATUS_TERMINATED && currState != protos.FlowStatus_STATUS_TERMINATING {
+				if currState == protos.FlowStatus_STATUS_COMPLETED {
+					changeErr = h.shutdownFlow(ctx, req.FlowJobName, req.DropMirrorStats, req.SkipDestinationDrop)
```

Comment:
why are we doing this? shouldn't we only need to delete from catalog?

Reply:
old code path for completed workflows deletes records synchronously since signals can't be used in this case